course/viewbase.py

```diff
@@ -117,6 +123,20 @@ def get_course_instance_object(self):
             course__url=self.kwargs[self.course_kw],
         )
 
+    def handle_exception(self, exc):
+        if isinstance(exc, TranslationNotFound):
+            instance_languages = self.instance.language
+            path_info = self.request.META.get('PATH_INFO')
```

Comment:
Maybe a bit nicer would be calling `request.get_full_path()`. Ouh, and `PATH_INFO` doesn't contain the full url when the app is bound to a sub path (see `SCRIPT_NAME` and `PATH_INFO` [here](https://wsgi.readthedocs.io/en/latest/definitions.html)).